### PR TITLE
Necro Sect - New Rite, Speed and Cost Tweaks, Bugfix

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request

After seeing some ingame usage of the sect, I've realised that the rites take forever and a day
So this reduces all the Necro Rite speeds and costs
It also adds the smoke spell to summoned undead as the chaplain starts with the book for that anyway.

Also at the request of @FortSumter and after playing a good few rounds myself I've decided to add a new rite to help entice the living crew to help out more - Create Lesser Lich
Grants one person the new lesser lichdom spell
- Higher respawn timer, Longer stun after revive, only two revives before the lesser phylactery breaks, doesnt spawn any items or gear on usage or resurrection

Also, I forgot to add the refund-favor section into the summon undead, so now thats in place too incase nobody takes the summon.

## Why It's Good For The Game

Less standing around, but not by much
No more favor loss if ghosts dont take the summon prompt
Summoned undead now have the chaplains smoke spell too

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image
image
image
image
image
image
image

https://user-images.githubusercontent.com/17776299/182030568-68ff78ee-9fb1-49c3-892e-6eeca0cbb392.mp4

</details>

## Changelog
:cl:
add: Necro Summoned Undead are now able to cast a smoke spell, just like the chaplain himself!
tweak: Necro Sect rites are now faster to perform and cost a bit less.
tweak: Necro Sect gains favor a little easier
fix: No more favor loss if the Neco Sect's Summon Undead rite fails.
add: Necro Sect now has a Create Lesser Lich rite which grants the new lesser lichdom spell
/:cl:

### CODE REVIEW

File: code/modules/religion/rites.dm [hunk 1]
```diff
@@ -277,6 +277,7 @@
 		to_chat(user, "<span class='warning'>The soul pool is empty...")
 		new /obj/effect/gibspawner/human/bodypartless(altar_turf)
 		user.visible_message("<span class='warning'>The soul pool was not strong enough to bring forth the undead.")
+		GLOB.religious_sect?.adjust_favor(1500, user) //refund if nobody takes the role
```
Comment (Sarchutar) [inline code/modules/religion/rites.dm]: couldn't one do `adjust_favor(favor_cost, user)` instead of having to set it manually?

Comment (ro5490) [reply code/modules/religion/rites.dm]: huh, never really thought about that
It *should* work

**Comment by ro5490** [comment]
As it is you can order monkey cube crates or get xenobio to supply slimes or cubes, if people want to RP to try and get the stations pets thats on them, but they shouldnt be breaking into anywhere to do so

**Comment by Sapfearon** [comment]
One more thing - this sect is kinda lacking....anything for chaplain or people who is here to worship cult. Flameproofing is for everyone, synth race too. Here you can't offer anything aside from watching at ritual of summoning.  
Maybe add 1 more rite which you can use on living crew? Can be anything - race change, funny spells, clothing, fitting weapons/armor, some kind of organs which you can use in hand? ( demon heart? armblade heart?) 
Just something to make people come and help chaplain with sacrifices - it's a bit sad when you can't offer anything to people who help you.

**Comment by FortSumter** [comment]
Played with the sect a few times now. Its ass. The sacrifice rite takes an infuriatingly long time to complete. For most passive creatures, you get so little favor points its laughable. 

You suggest monkies or slimes, but monkies will move and unbuckle from the altar and only net you 200 each out of the 1500 points needed to do one ritual. I haven't tried slimes yet but from what I understand they're hostile to crew who haven't been feeding them. The most points I got out of rituals were mindless gondolas, which were worth 500 points each. Hostile mobs have the most health and therefore grant the most favor, but good luck getting them on the altar and not having it kill you while you sit there for 5 minutes doing the sacrifice rite.

The rewards for doing all this and killing dozens of passive mobs is quite lackluster as well. You can create a holy undead for 1500 favor and while it got me giddy that they can bless things like the chaplain can, I found them otherwise useless as a non-antag. Reviving someone for 2500 favor is insanely expensive for something medical does all the time for free. Definitely not worth.

**Comment by ro5490** [comment]
> Maybe add 1 more rite which you can use on living crew?

Sounds good, I'll look into it and see what works best here



> The sacrifice rite takes an infuriatingly long time to complete.

This is being patched with this PR



> For most passive creatures, you get so little favor points its laughable.

As is this, I just forgot to push the changes

**Comment by FortSumter** [comment]
Thank you for taking suggestions on this as I genuinely do appreciate it. I don't think lichdom is a good idea though...

I can't think of many things more broken for normal crew to have than either Blood Crawl or Lichdom, and for similar reasons.  I get that you chose lichdom because it's thematic, but imagine the captain getting it and binding their soul to something they put on the AI sat or some other god-forsaken place only he can access with ease. You could only kill him temporarily and he would know who it was who killed him meaning if he was your assassination objective you're screwed. Alternatively, imagine a traitor becoming a lich and mass producing chemical grenades. As a skeleton he isn't affected by heat or fire and so he could just burn the station down if he liked or more hilarious he could suicide bomb whomever he liked with the knowledge that it will not be the end of him.

I'm not sure what to suggest instead except maybe just transforming people into skeletons as just being a skeleton could be pretty fun and it'd be in line with what other sects do.

**Comment by ro5490** [comment]
Tempted to keep lichdom as a super expensive rite, say like a full 10k favor or something
And add a race-change rite to skeleton/zombie perhaps

Cant think of anything else that would really fit the sect right now
Want some input on this first though

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
This is just overpowered as hell, this is a wizard spell that makes you basically immortal and is designed for someone who is doing a 1v79 rather than the random chaplain who is probably going 1v1 against some random antag at most.

There needs to be a new version of the item with WAY more drawbacks:
 - limited use/cooldown
 - longer revive timers
 - unable to be revived through any other source

**Comment by ro5490** [comment]
I'll just make it a race change to skeleton instead tbh, thanks ^-^

**Comment by PowerfulBacon** [comment]
Use the devil hellbound var, it prevents revival

**Comment by ro5490** [comment]
all set ^-^

**Comment by JukeJukestar** [comment]
> You suggest monkies or slimes, but monkies will move and unbuckle from the altar and only net you 200 each out of the 1500 points needed to do one ritual. 

Just wanted to add, sacrificing monkeys is rather easy if you do it right.
After you get it from the monkey cube, hit it with your null rod until its horizontal.
After that its really easy to sacrifice it and you have a pretty good timeframe before it dies to bloodloss in crit so you are always able to complete the ritual without it dieing halfway through

**Comment by FortSumter** [comment]
You're saying that like it wasn't what I was doing. Idk if the points are related to max health the animal had or how much it had when it died (probably the former), but monkies will fight back and having to go to med for heals between killing 1-2 monkies is cringe for only 200 points out of 1500 and also having a chance for the monkey to die before the ritual is completed depending on if you were interrupted is also cringe. On top of that, you don't get monkey meat for sacrificing one like you do any other pet, so you can't even do the rites just to supply the cook with meats.

**Comment by ro5490** [comment]
It's down to the current hp of the creature, so if you've put them in crit you get less points,  best way I've found for monkeys is spawn one, chairbuckle and cable cuff quickly, then hook them up to the altar, they still sometimes fight back but with these changes you'll get more favor too

File: code/modules/religion/rites.dm [hunk 2]
```diff
@@ -255,17 +255,94 @@
...
+	name = "Create Lesser Lich"
+	desc = "Gives the bound creature a spell granting them the ability to create a lesser phylactery, causing them to become a skeleton and revive on death twice if the phylactery still exists on-station. Be warned, becoming a lesser lich will prevent revivial by any other means."
+	ritual_length = 70 SECONDS
+	ritual_invocations = list("From the depths of the soul pool ...",
+	"... come forth into this being ...",
+	"... grant this servant power ...",
+	"... grant them temporary immortality ...")
+	invoke_msg = "... Grant them the power to become one with necromancy!!"
+	favor_cost = 2250
+//the creature chosen for the rite
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: the comments should be indented and formatted like the following:
```
/// Comment here
```
The three slashes are necessary. Also, describe more ("the spell" -> "The typepath of the spell to grant")

File: code/modules/religion/rites.dm [hunk 3]
```diff
@@ -255,17 +255,94 @@
...
+	var/lichspell = /obj/effect/proc_holder/spell/targeted/lesserlichdom
+
+/datum/religion_rites/create_lesser_lich/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user,"<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
+	if(LAZYLEN(movable_reltool.buckled_mobs))
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: `length()`

File: code/modules/religion/rites.dm [hunk 4]
```diff
@@ -255,17 +255,94 @@
...
+	var/mob/living/lich_to_be
+//the spell
+	var/lichspell = /obj/effect/proc_holder/spell/targeted/lesserlichdom
+
+/datum/religion_rites/create_lesser_lich/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user,"<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: remove this check

File: code/modules/religion/rites.dm [hunk 5]
```diff
@@ -255,17 +255,94 @@
...
+			lich_to_be = null
+			return FALSE
+		for(var/obj/effect/proc_holder/spell/knownspell in lich_to_be.mob_spell_list)
+			if(knownspell.type == lichspell)
+				to_chat(user,"<span class='warning'>You've already empowered [lich_to_be], get them to use the spell granted to them! To empower another, they must be buckled to [movable_reltool].</span>")
+				lich_to_be = null
+				return FALSE
+		to_chat(user,"<span class='warning'>You're going to empower the [lich_to_be] who is buckled on [movable_reltool].</span>")
+		return ..()
+	else
```
Comment (itsmeow) [inline code/modules/religion/rites.dm:292]: chain all of this into else if blocks, because they all return

File: code/modules/religion/rites.dm [hunk 6]
```diff
@@ -255,17 +255,94 @@
...
+		to_chat(user,"<span class='warning'>You're empowering yourself!</span>")
+		return ..()
+
+
+/datum/religion_rites/create_lesser_lich/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	..()
+	if(!ismovable(religious_tool))
+		CRASH("[name]'s perform_rite had a movable atom that has somehow turned into a non-movable!")
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool?.buckled_mobs?.len)
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: `!length(movable_reltool.buckled_mobs)`

File: code/modules/religion/rites.dm [hunk 7]
```diff
@@ -255,17 +255,94 @@
...
+		lich_to_be = user
+	else
+		for(var/buckled in movable_reltool.buckled_mobs)
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: `var/mob/living/human/buckled in` instead of the ishuman

File: code/modules/religion/rites.dm [hunk 8]
```diff
@@ -255,17 +255,94 @@
...
+			if(ishuman(buckled))
+				lich_to_be = buckled
+				break
+	if(!lich_to_be)
+		return FALSE
+	lich_to_be.AddSpell(new /obj/effect/proc_holder/spell/targeted/lesserlichdom(null))
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: this should be a new `lichspell` instead of typing out the typepath

File: code/modules/spells/spell_types/lesserlichdom.dm [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "skeleton"
+
+/obj/effect/proc_holder/spell/targeted/lesserlichdom/cast(list/targets,mob/user = usr)
+	for(var/mob/M in targets)
+		var/list/hand_items = list()
+		if(iscarbon(M))
+			hand_items = list(M.get_active_held_item(),M.get_inactive_held_item())
+		if(!hand_items.len)
```
Comment (itsmeow) [inline code/modules/spells/spell_types/lesserlichdom.dm]: `length(hand_items)`

File: code/modules/spells/spell_types/lesserlichdom.dm [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+		if(!M.mind.hasSoul)
+			to_chat(user, "<span class='caution'>You do not possess a soul.</span>")
+			return
+
+		var/obj/item/marked_item
+
+		for(var/obj/item/item in hand_items)
+			// I ensouled the nuke disk once. But it's probably a really
+			// mean tactic, so probably should discourage it.
+			if((item.item_flags & ABSTRACT) || HAS_TRAIT(item, TRAIT_NODROP) || SEND_SIGNAL(item, COMSIG_ITEM_IMBUE_SOUL, user))
```
Comment (itsmeow) [inline code/modules/spells/spell_types/lesserlichdom.dm:40]: does this block soul shards if they already have a soul?

File: code/modules/spells/spell_types/lesserlichdom.dm [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+/obj/item/lesserphylactery/Initialize(mapload, datum/mind/newmind)
+	. = ..()
+	mind = newmind
+	name = "lesser phylactery of [mind.name]"
+
+	active_phylacteries++
+	GLOB.poi_list |= src
+	START_PROCESSING(SSobj, src)
+	if(initial(SSticker.mode.round_ends_with_antag_death))
+		SSticker.mode.round_ends_with_antag_death = FALSE
```
Comment (itsmeow) [inline code/modules/spells/spell_types/lesserlichdom.dm]: what's the point of this?

File: code/modules/spells/spell_types/lesserlichdom.dm [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+
+	lich.real_name = mind.name
+	mind.transfer_to(lich)
+	mind.grab_ghost(force=TRUE)
+	lich.hardset_dna(null,null,lich.real_name,null, new /datum/species/skeleton,null)
+	to_chat(lich, "<span class='warning'>Your bones clatter and shudder as you are pulled back into this world!</span>")
+	var/turf/body_turf = get_turf(old_body)
+	lich.Paralyze(400 + 200*resurrections) // paralyzed for longer due to lesser spell
+	resurrections++
+	if(old_body?.loc)
```
Comment (itsmeow) [inline code/modules/spells/spell_types/lesserlichdom.dm:129]: early return this

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/religion/rites.dm (hunk 5 above)
Comment (ro5490) [reply code/modules/religion/rites.dm:292]: little confused on what specifically needs to be else-if here, 294?

Comment (itsmeow) [reply code/modules/religion/rites.dm:292]: all of it can be chained else ifs

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 11 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm]: Spell is also in the wizard spellbook, I'll add a check for if its an antag so regular chaplains dont end up toggling this

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 10 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm:40]: If the original lichdom spell does, this should as well, but I dont believe it does

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 12 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm:129]: I have no idea what you mean, sorry

Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm:129]: if(!x)
   return

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 11 above)
Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm]: I am confused what this toggle does and why you are toggling it

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 10 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm:40]: It seems neither the original wizard spell, or this weaker version blocks soul shards

Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm:40]: Do soul shards return false to COMSIG IMBUE SOUL if they already have one?

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 11 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm]: Its the round end toggle, if the round is meant to toggle on antag death, it stops the round ending, as the lich would technically be dead but awaiting phylactery revival.
I've put in a special_role check so this only toggles if an antag creates a phylatery.
tl;dr - no round end on antag death, toggles back to end on antag death if phylactery gets destroyed

Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm]: > Spell is also in the wizard spellbook, I'll add a check for if its an antag so regular chaplains dont end up toggling this

This I believe is for the wizard gamemode, where if the lich dies the round ends. Right?

Lesser lichdom should NEVER toggle this.

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 12 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm:129]: for 	
if(old_body?.loc)
?
(this is pretty much just a copy of the original lich spell code)

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 10 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm:40]: Nope, seems they just add a soul to contents, and if contents contains anything then they're used, or something like that

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 11 above)
Comment (ro5490) [reply code/modules/spells/spell_types/lesserlichdom.dm]: I'll remove lesser lichdom from wizard spellbook, and remove the toggle

File: code/modules/religion/rites.dm (hunk 5 above)
Comment (ro5490) [reply code/modules/religion/rites.dm:292]: this is gonna sound really dumb but I dont get what you mean as this is what I'm looking at 
image

do you mean like this, because to me this seems wrong, without even trying to compile or test...
image

File: code/modules/spells/spell_types/lesserlichdom.dm [hunk 13]
```diff
@@ -97,6 +97,7 @@
 	STOP_PROCESSING(SSobj, src)
 	active_phylacteries--
 	GLOB.poi_list -= src
+	. = ..()
```
Comment (itsmeow) [inline code/modules/spells/spell_types/lesserlichdom.dm]: return ..() is cleaner

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 12 above)
Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm:129]: remove the nesting.
Put before the rest of the code
```
if(!old_body?.loc)
    return
```

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 10 above)
Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm:40]: Alright, well, unrelated change. Just leave it for now - or if you want you can register the IMBUE_SOUL comsig on soul shards and refactor them to deny it if they have a soul.

File: code/modules/religion/rites.dm (hunk 5 above)
Comment (itsmeow) [reply code/modules/religion/rites.dm:292]: oh, guh, mobile formatting. sorry. it's good as it is

File: code/modules/religion/rites.dm (hunk 7 above)
Comment (itsmeow) [reply code/modules/religion/rites.dm]: A typed loop, is what I mean
```
for(var/mob/living/human/buckled in movable_reltool.buckled_mobs)
    ...
```

File: code/modules/spells/spell_types/lesserlichdom.dm (hunk 12 above)
Comment (itsmeow) [reply code/modules/spells/spell_types/lesserlichdom.dm:129]: the github UI has once again ruined my ability to see formatting, it should be fine

File: code/modules/religion/rites.dm [hunk 14]
```diff
@@ -253,19 +253,90 @@
...
+
+/datum/religion_rites/create_lesser_lich/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	..()
+	if(!ismovable(religious_tool))
+		CRASH("[name]'s perform_rite had a movable atom that has somehow turned into a non-movable!")
+	var/atom/movable/movable_reltool = religious_tool
+	if(!length(movable_reltool.buckled_mobs))
+		lich_to_be = user
+	else
+		for(var/buckled in movable_reltool.buckled_mobs)
```
Comment (itsmeow) [inline code/modules/religion/rites.dm]: unaddressed still

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/religion/rites.dm (hunk 14 above)
Comment (ro5490) [reply code/modules/religion/rites.dm]: ah, it was a hidden reply for some reason so I missed it

**Review by itsmeow — DISMISSED** [review]

**Comment by itsmeow** [comment]
lint machine broke

```
code/modules/religion/rites.dm, line 318, column 3:
error: undefined type: /mob/living/human
```

it's `/mob/living/carbon/human`

**Comment by ro5490** [comment]
whoops

**Review by itsmeow — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]
